integration-tests/deployment/ccip/deploy_home_chain.go

```diff
@@ -642,6 +616,278 @@ func setupExecDON(
 	return nil
 }
 
+func promoteCandidateExecOps(
```

Comment:
Quite a lot of duplicate logic between `promoteCandidate___Ops` and `setCandidate___Ops`, where the only difference is the plugin config we pass, and whether we call `AddDon` or `UpdateDon`.  We can simplify this a lot with a `promoteCandidatePluginOp` like in my original PR that accepts either an execConfig or commitConfig (plugin type information is included).

Since it's not necessarily true that we always add the commit plugin first, it doesn't make sense to associate that with `setCandidateCommitOps` either.

Reply:
it should be able to accept just a CCIPHomeOCR3Config which is plugin agnostic and apply that